worlds/mm2/__init__.py

```diff
@@ -0,0 +1,314 @@
+            if self.options.yoku_jumps:
+                robot_master_pool.append(0)
+            if self.options.enable_lasers:
+                robot_master_pool.append(4)
+            self.options.starting_robot_master.value = self.random.choice(robot_master_pool)
+            logger.warning(
+                f"Mega Man 2 ({self.player_name}): "
+                f"Incompatible starting Robot Master, changing to "
+                f"{self.options.starting_robot_master.current_key.replace('_', ' ').title()}")
+
+    def generate_basic(self) -> None:
+        goal_location = self.multiworld.get_location(dr_wily, self.player)
+        goal_location.place_locked_item(MM2Item("Victory", ItemClassification.progression, None, self.player))
+        self.multiworld.completion_condition[self.player] = lambda state: state.has("Victory", self.player)
+
+    def fill_hook(self,
+                  progitempool: List["Item"],
+                  usefulitempool: List["Item"],
+                  filleritempool: List["Item"],
+                  fill_locations: List["Location"]) -> None:
+        # on a solo gen, fill can try to force Wily into sphere 2, but for most generations this is impossible
+        # since MM2 can have a 2 item sphere 1, and 3 items are required for Wily
+        if self.multiworld.players > 1:
+            return  # Don't need to change anything on a multi gen, fill should be able to solve it with a 4 sphere 1
+        rbm_to_item = {
+            0: heat_man_stage,
+            1: air_man_stage,
+            2: wood_man_stage,
+            3: bubble_man_stage,
+            4: quick_man_stage,
+            5: flash_man_stage,
+            6: metal_man_stage,
+            7: crash_man_stage
+        }
+        affected_rbm = [2, 3]  # Wood and Bubble will always have this happen
+        possible_rbm = [1, 5]  # Air and Flash are always valid targets, due to Item 2/3 receive
+        if self.options.consumables:
+            possible_rbm.append(6)  # Metal has 3 consumables
+            possible_rbm.append(7)  # Crash has 3 consumables
+            if self.options.enable_lasers:
+                possible_rbm.append(4)  # Quick has a lot of consumables, but needs logical time stopper if not enabled
+        else:
+            affected_rbm.extend([6, 7])  # only two checks on non consumables
+        if self.options.yoku_jumps:
+            possible_rbm.append(0)  # Heat has 3 locations always, but might need 2 items logically
+        if self.options.starting_robot_master.value in affected_rbm:
+            rbm_names = list(map(lambda s: rbm_to_item[s], possible_rbm))
+            valid_second = [item for item in progitempool
+                            if item.name in rbm_names
+                            and item.player == self.player]
+            placed_item = self.random.choice(valid_second)
+            rbm_defeated = f"{self.options.starting_robot_master.get_option_name(self.options.starting_robot_master.value)} - Defeated"
```

Comment:
This line really does just look ridiculous, and I'm too tired to go figure out why now

Reply:
Is this not just `option.current_key`?

Reply:
No, get_option_name returns the formalized version of the current key. So when current_key is `heat_man` it returns `Heat Man`

Reply:
Why not use `auto_display_name` ?

Reply:
I genuinely feel like this is clearer, which, maybe I'm crazy for thinking it but 🤷 
`self.options.starting_robot_master.current_key.replace('_', ' ').title()`

Reply:
No point reinventing the wheel if the function already exists. I think `get_current_option_name` existed at some point but doesn't anymore?

Reply:
You can use `self.options.starting_robot_master.current_option_name`

Reply:
Medic is right, and you can see the code of `Option.current_option_name`, it's literally just what you have here:

```
    @property
    def current_option_name(self) -> str:
        """For display purposes. Worlds should be using current_key."""
        return self.get_option_name(self.value)
```

I will only allow this (the old code **or** Medic's suggestion), though, if location names are not used for logic anywhere. Both of the functions in question (`.get_option_name` and `.current_option_name`) are for display purposes only. Worlds and clients that play nice should generally not use location names for program logic anyway, so I'd be willing to allow it.

Reply:
(This is technically "logic", but the second I change it I'll know it breaks because it only runs on solo gens). That being said, I'm pretty sure I have the relevant mapping in another file anyways I can just import.